### PR TITLE
Bug/teardown podman

### PR DESCRIPTION
**Pull request by l0r1s**
Fix https://github.com/paritytech/zombienet/issues/757

* Fixed podman issue when trying to teardown before fully spawned.
* Fixed podman issue when the network is spawned to prevent exiting zombienet

### CODE REVIEW

File: javascript/packages/cli/src/actions/spawn.ts [hunk 1]
```diff
@@ -93,8 +94,10 @@ export async function spawn(
     force,
     inCI,
     silent: false,
+    setGlobalNetwork,
   };
   const network = await start(creds, config, options);
   network.showNetworkInfo(config.settings?.provider);
-  return network;
+  // keep the process running
+  await new Promise(() => setInterval(() => {}, 1000));
```
Comment (pepoviola) [inline javascript/packages/cli/src/actions/spawn.ts:102]: I think we don't need a promise here.
```suggestion
setInterval(() => {}, 1000);
```

File: javascript/packages/orchestrator/src/orchestrator.ts [hunk 2]
```diff
@@ -161,6 +162,9 @@ export async function start(
     if (networkSpec.settings.node_spawn_timeout)
       client.timeout = networkSpec.settings.node_spawn_timeout;
     network = new Network(client, namespace, tmpDir.path);
+    if (options?.setGlobalNetwork) {
+      options.setGlobalNetwork(network);
+    }
```
Comment (pepoviola) [inline javascript/packages/orchestrator/src/orchestrator.ts:167]: I think you can just do
```suggestion
    options?.setGlobalNetwork(network);
```

File: javascript/packages/orchestrator/src/providers/podman/podmanClient.ts [hunk 3]
```diff
@@ -297,10 +300,13 @@ export class PodmanClient extends Client {
...
-        `\n ${decorators.red("Error: ")} \t ${decorators.bright(error)}\n`,
-      );
-      throw error;
+      if (!this.isTearingDown) {
+        console.log(
+          `\n ${decorators.red("Error: ")} \t ${decorators.bright(error)}\n`,
+        );
+        throw error;
+      }
+      return { exitCode: 0, stdout: "" };
```
Comment (pepoviola) [inline javascript/packages/orchestrator/src/providers/podman/podmanClient.ts:309]: We can use the same pattern that we already use in the `native` provider, and set `allowFail` in the caller.

```suggestion
      debug(error);
      if (!opts?.allowFail) throw error;

      const { exitCode, stdout, message: errorMsg } = error;

      return {
        exitCode,
        stdout,
        errorMsg,
      };
```

File: javascript/packages/orchestrator/src/orchestrator.ts (hunk 2 above)
Comment (l0r1s) [reply javascript/packages/orchestrator/src/orchestrator.ts:167]: Just tried and Typescript gives me:

```Cannot invoke an object which is possibly 'undefined'.```

Comment (l0r1s) [reply javascript/packages/orchestrator/src/orchestrator.ts:167]: Got ```Property 'b' does not exist on type 'never'.``` if I try to copy the code in the project to use our TS config

File: javascript/packages/orchestrator/src/providers/podman/podmanClient.ts (hunk 3 above)
Comment (l0r1s) [reply javascript/packages/orchestrator/src/providers/podman/podmanClient.ts:309]: This part is for handling the errors possibly thrown by previous call of runCommand, not the actual calls in the destroyNamespace. In that case, it keeps the behavior of throwing if we are not tearing down the network.

**Review by pepoviola — APPROVED** [review]

**Comment by pepoviola** [comment]
thanks!!

**Pull request merged**